bgpd/bgp_vty.c

```diff
@@ -11454,7 +12562,72 @@ DEFUN (show_ip_bgp_lcommunity_info,
 
 	return CMD_SUCCESS;
 }
+/* Graceful Restart */
+
+static void bgp_show_global_graceful_restart_mode_vty(struct vty *vty,
+					struct bgp *bgp,
+					bool use_json,
+					json_object *json)
+{
+
+
+	vty_out(vty, "\n%s", SHOW_GR_HEADER);
 
+	int bgp_global_gr_mode = bgp_global_gr_mode_get(bgp);
+
+	switch (bgp_global_gr_mode) {
+
+	case GLOBAL_HELPER:
+		vty_out(vty,
+				"Global Bgp GR Mode :  Helper\n");
+		break;
+
+	case GLOBAL_GR:
+		vty_out(vty,
+			"Global Bgp GR Mode :  Restart\n");
+		break;
+
+	case GLOBAL_DISABLE:
+		vty_out(vty,
+				"Global Bgp GR Mode :  Disable\n");
+		break;
+
+	case GLOBAL_INVALID:
+	default:
+		vty_out(vty,
+			"Global Bgp GR Mode  Invalid\n");
+		break;
+	}
+	vty_out(vty, "\n");
+}
+
+static int  bgp_show_neighbor_graceful_restart_afi_all(struct vty *vty,
+						enum show_type type,
+						const char *ip_str,
+						afi_t afi,
+						bool use_json)
+{
+	if ((afi == AFI_MAX) && (ip_str == NULL)) {
+		afi = AFI_IP;
+
+		while ((afi != AFI_L2VPN) && (afi < AFI_MAX)) {
```

Comment:
`afi < AFI_MAX` will be always valid due to `afi = AFI_IP;`

Reply:
its a while loop, it would start at afi = AFI_IP and then afi++. so the boundary check.